external-crates/move/crates/move-disassembler/src/disassembler.rs

```diff
@@ -1026,32 +1285,38 @@ impl<'a> Disassembler<'a> {
             | Bytecode::MoveFromGenericDeprecated(_)
             | Bytecode::MoveToDeprecated(_)
             | Bytecode::MoveToGenericDeprecated(_) => {
-                any_write!(buffer, "DEPRECATED BYTECODE: {instruction:?}")
+                any_write!(buffer, "DEPRECATED BYTECODE: {instruction:?}").map(|_| ())
             }
             // All other instructions are OK to be printed using the standard debug print.
-            x => any_write!(buffer, "{x:#?}"),
-        }
+            x => any_write!(buffer, "{x:#?}").map(|_| ()),
+        }?;
+        let inst_end_offset = buffer.byte_len();
+        Ok(Loc::new(
+            FileHash::empty(),
+            inst_start_offset,
+            inst_end_offset,
+        ))
     }
 
     // These need to be in the context of a function or a struct definition since type parameters
     // can refer to function/struct type parameters.
     fn disassemble_sig_tok(
         &self,
-        buffer: &mut impl Write,
+        buffer: &mut (impl Write + ByteLength),
         sig_tok: &SignatureToken,
         type_instantiation: Option<&[SignatureToken]>,
         type_param_name_context: &[SourceName],
     ) -> Result<()> {
         match sig_tok {
-            SignatureToken::Bool => any_write!(buffer, "bool"),
-            SignatureToken::U8 => any_write!(buffer, "u8"),
-            SignatureToken::U16 => any_write!(buffer, "u16"),
-            SignatureToken::U32 => any_write!(buffer, "u32"),
-            SignatureToken::U64 => any_write!(buffer, "u64"),
-            SignatureToken::U128 => any_write!(buffer, "u128"),
-            SignatureToken::U256 => any_write!(buffer, "u256"),
-            SignatureToken::Address => any_write!(buffer, "address"),
-            SignatureToken::Signer => any_write!(buffer, "signer"),
+            SignatureToken::Bool => any_write!(buffer, "bool").map(|_| ()),
```

Comment:
IMO would be cleaner to do something like 
```
match sig_tok {
 ....
}.map(|_| ())
```

or to `?` all of these and then have an `Ok(())` at the bottom of the function.

Reply:
I tried that some match arms return `Result<()>` instead of `Result<Loc>`, and I had to unify them which made this look uglier...